IdentityCore/src/cache/crypto/MSIDSymmetricKey.m

```diff
@@ -175,13 +175,13 @@ - (uint8_t *)kdfCounterMode:(uint8_t *)keyDerivationKey
 
 
 /*
- *Function used to shift data of 1 byte. This byte is the "ctr".
+ *Function used to shift data by 4 byte and insert ctr in the first 4 bytes.
  */
-- (uint8_t *)updateDataInput:(uint8_t)ctr
+- (uint8_t *)updateDataInput:(uint32_t)ctr
                   fixedInput:(uint8_t *)fixedInput
            fixedInput_length:(size_t)fixedInput_length
 {
-    uint8_t *tmpFixedInput = (uint8_t *)malloc(fixedInput_length + 4); //+4 is caused from the ct
+    uint8_t *tmpFixedInput = (uint8_t *)malloc(fixedInput_length + sizeof(uint32_t)); //+4 is caused from the ct
```

Comment:
It's unneccessary as ctr will be 0000 0000 0000 0001
    uint8_t *tmpFixedInput = (uint8_t *)malloc(fixedInput_length + 4); //+4 is caused from the ct

tmpFixedInput = 0x00 0x00 0x00 0x00 (fixedInput_length)

ctr value is very small as in spec (L =256 / h = 256) => ctr = 1 so uint32_t or uint8_t will not make any different in the calcuate (ctr >> 24) or (ctr>>16) or (ctr >> 8).